src/main/java/nl/recognize/dwh/application/loader/AbstractEntityLoader.java

```diff
@@ -175,28 +175,30 @@ private Object mapField(Object entity, FieldMapping field, List<RequestFilter> u
             value = ((ZonedDateTime) value).toOffsetDateTime().toString();
         }
 
-        if (Arrays.asList(FieldMapping.TYPE_ENTITY, FieldMapping.TYPE_LIST).contains(type)) {
+        if (Arrays.asList(FieldMapping.TYPE_ENTITY, FieldMapping.TYPE_LIST, FieldMapping.TYPE_SET).contains(type)) {
             Mapping mapping = field.getEntryMapping();
 
             if (!(mapping instanceof EntityMapping) && !(mapping instanceof FieldMapping)) {
                 throw new IllegalStateException(String.format("Invalid entity mapping for collection at field %s", name));
             }
 
-            if (type.equals(FieldMapping.TYPE_LIST)) {
-                if (value == null) {
-                    return new ArrayList<>();
-                }
-                List<Object> values = (List<Object>) value;
-                return values.stream()
+            Collection<Object> values = new ArrayList<>();
+            if (value != null) {
+                if (FieldMapping.TYPE_SET.equals(type) || FieldMapping.TYPE_LIST.equals(type)) {
+                    values = (Collection<Object>) value;
+                    return values.stream()
                         .map(aValue -> mapping instanceof EntityMapping
                                 ? mapEntity(aValue, (EntityMapping) mapping, usedFilters)
                                 : mapField(aValue, (FieldMapping) mapping, usedFilters)
                         ).collect(Collectors.toList());
-            } else {
+                }
+
                 return mapping instanceof EntityMapping
                         ? mapEntity(value, (EntityMapping) mapping, usedFilters)
                         : mapField(value, (FieldMapping) mapping, usedFilters);
             }
+
+            return null;
```

Comment:
Dit veranderd zo te zien het gedrag voor `value == null` en `TYPE_LIST`. In de oorspronkelijke code wordt dan een lege lijst terug gegeven.

Verder wat vragen:

Moet de set ook echt een set zijn? Want dat is nu niet per se het geval. Zo nee, zou het dan niet makkelijker zijn om Collections naar het reeds bestaande list te mappen? Nu heb je namelijk al dat voor set en list in principe dezelfde mapping plaatsvind.

Reply:
Ik heb geen flauw idee wat er verwacht wordt, er zijn namelijk geen testen voor dit stuk code

Reply:
je mag er een test bij maken :-)